specs/trust-store-trust-policy.md

```diff
@@ -301,6 +388,29 @@ Signature verification levels provide defined behavior for each validation e.g.
     }
 ```
 
+#### Timestamp Countersignature Verification details
+
+ Sufficient and necessary conditions to trigger timestamp countersignature verification:
+  - At least one certificate in the signing certificate chain has expired at the time of verification.
+  - The signing scheme is [`notary.x509`](./signing-scheme.md/#notaryx509).
+  - The unsigned attribute `Timestamp Signature` is present in the signature envelope.
+  - The `authenticTimestamp` under `signatureVerification` of trust policy is NOT marked as `skip`.
+  - `level` under `timestampVerification` of trust policy is set to `strict` or `timestampVerification` is not present, default values are used in this case.
+   
+ Timestamp countersignature verification is a multi step process performs the following validations
+  - integrity (timestamp token is unaltered, countersignature is not corrupted)
+  - authenticity (the timestamp token is really from the TSA that claims to have signed it)
+  - expiry (the timestamp certificate chain is unexpired at the time of verification)
+  - revocation check (is the TSA still trusted at the time of verification)
+
+ **Integrity** : Guarantees that the timestamp token wasn't altered after it was signed, or the countersignature isn't corrupted. It is always enforced if timestamp countersignature verification is performed.
+
+ **Authenticity** : Guarantees that the timestamp was issued by a trusted TSA identity. Its definition does not include revocation, which is when a trusted TSA is subsequently untrusted because of a compromise. It is always enforced if timestamp countersignature verification is performed.
+
+ **Expiry** : Guarantees that the timestamp certificate chain is unexpired at the time of verification. It can be relaxed through the `expiryRelaxed` option under `timestampVerification` of the trust policy. When `expiryRelaxed` is set to `true`, the timestamp certificate chain expiry is checked against the time point been stamped. Otherwise, the timestamp certificate chain expiry is checked against the time at verification.
```

Comment:
Suggestion: When `expiryRelaxed` is set to `true`, the timestamp in the timestamp signature is trusted as the current time to verify the timestamp signature itself.